cms/djangoapps/contentstore/views/videos.py

```diff
@@ -338,11 +340,13 @@ def transcript_preferences_handler(request, course_key_string):
 
     Returns: valid json response or 400 with error message
     """
+    transcript_preferences = None
     data = request.json
-    provider = data.get('provider', '')
+    provider = data.get('provider', None)
 
-    # TODO: if provider == '': delete course preferences
-    # i.e call delete api end point like delete_transcript_preferences(course_key_string)
+    if not provider:
```

Comment:
Can we introduce something more like an `action`? action can be create/delete

Reply:
action ?

Reply:
@muzaffaryousaf Should we go for `request.method == 'POST':` and `request.method == 'DELETE':` actions approach here ? or we are good to go here ?

Reply:
I think so.